src/main/java/be/seeseemelk/mockbukkit/block/state/LootableContainerMock.java

```diff
@@ -0,0 +1,208 @@
+package be.seeseemelk.mockbukkit.block.state;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+import be.seeseemelk.mockbukkit.UnimplementedOperationException;
+import com.destroystokyo.paper.loottable.LootableBlockInventory;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.loot.LootTable;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+public abstract class LootableContainerMock extends ContainerMock implements LootableBlockInventory
+{
+
+	private boolean refillEnabled = false;
+	private long lastFilled = -1;
+	private long nextRefill = -1;
+
+	private Map<UUID, Long> lootedPlayers;
+
+	/**
+	 * Constructs a new {@link LootableContainerMock} for the provided {@link Material}.
+	 *
+	 * @param material The material this state is for.
+	 */
+	protected LootableContainerMock(@NotNull Material material)
+	{
+		super(material);
+	}
+
+	/**
+	 * Constructs a new {@link LootableContainerMock} for the provided {@link Block}.
+	 *
+	 * @param block The block this state is for.
+	 */
+	protected LootableContainerMock(@NotNull Block block)
+	{
+		super(block);
+	}
+
+	/**
+	 * Constructs a new {@link ContainerMock} by cloning the data from an existing one.
+	 *
+	 * @param state The state to clone.
+	 */
+	protected LootableContainerMock(@NotNull ContainerMock state)
+	{
+		super(state);
+	}
+
+	@Override
+	public boolean isRefillEnabled()
+	{
+		return this.refillEnabled;
+	}
+
+	/**
+	 * Sets whether this container should be refilled.
+	 *
+	 * @param refillEnabled Whether this container should be refilled.
+	 */
+	public void setRefillEnabled(boolean refillEnabled)
+	{
+		this.refillEnabled = refillEnabled;
+	}
+
+	@Override
+	public boolean hasBeenFilled()
+	{
+		return getLastFilled() != -1;
+	}
+
+	@Override
+	public boolean canPlayerLoot(@NotNull UUID player)
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
+	@Override
+	public boolean hasPlayerLooted(@NotNull UUID player)
+	{
+		return this.lootedPlayers != null && this.lootedPlayers.containsKey(player);
+	}
+
+	@Override
+	public @Nullable Long getLastLooted(@NotNull UUID player)
+	{
+		return lootedPlayers != null ? lootedPlayers.get(player) : null;
+	}
+
+	@Override
+	public boolean setHasPlayerLooted(@NotNull UUID player, boolean looted)
+	{
+		final boolean hasLooted = hasPlayerLooted(player);
+
+		if (this.lootedPlayers == null)
+		{
+			this.lootedPlayers = new HashMap<>();
+		}
+
+		if (hasLooted != looted)
+		{
+			if (looted)
+			{
+				this.lootedPlayers.computeIfAbsent(player, p -> System.currentTimeMillis());
+			}
+			else
+			{
+				this.lootedPlayers.remove(player);
+			}
+		}
+		return hasLooted;
+	}
+
+	@Override
+	public boolean hasPendingRefill()
+	{
+		return nextRefill != -1;
```

Comment:
```suggestion
		return nextRefill != -1 && nextRefil > lastFilled;
```
This is what's done in paper. I don't see why we shouldn't follow that